app/(main)/leaderboard/agents/AgentsLeaderboardTable.tsx

```diff
@@ -0,0 +1,256 @@
+'use client';
+
+import { useState } from 'react';
+import {
+  Table,
+  TableBody,
+  TableCaption,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from '@/app/components/ui/Table';
+import { Button, Icon, IconButton } from '@swapr/ui';
+import { Avatar, Skeleton } from '@/app/components';
+import { useQuery } from '@tanstack/react-query';
+import { getAgentsLeaderboardData } from '@/queries/dune';
+import { twMerge } from 'tailwind-merge';
+import Link from 'next/link';
+import { TWELVE_HOURS_IN_MS } from '@/utils';
+
+const ITEMS_PER_PAGE = 100;
+
+enum SortOrder {
+  ASC = 'asc',
+  DESC = 'desc',
+}
+
+const SortKey = {
+  PROFIT_LOSS: 'profit_loss',
+  TOTAL_VOLUME: 'total_volume',
+  SUCCESS_RATE: 'success_rate',
+  TOTAL_WINS: 'total_wins',
+  TOTAL_POSITIONS: 'total_positions',
+} as const;
+
+type SortKeyType = (typeof SortKey)[keyof typeof SortKey];
+
+export default function AgentsLeaderboardTable() {
+  const [sortKey, setSortKey] = useState<SortKeyType>(SortKey.PROFIT_LOSS);
+  const [sortOrder, setSortOrder] = useState<SortOrder>(SortOrder.DESC);
+  const [page, setPage] = useState(1);
+
+  const { data, isLoading } = useQuery({
+    queryKey: ['getAgentsLeaderboardData', page, ITEMS_PER_PAGE, sortKey, sortOrder],
+    queryFn: () =>
+      getAgentsLeaderboardData({
+        page,
+        pageSize: ITEMS_PER_PAGE,
+        sort_by: `${sortKey} ${sortOrder}`,
+      }),
+    staleTime: TWELVE_HOURS_IN_MS,
+  });
+
+  const agentsLeaderboardData = data?.data ?? [];
+  const totalPages = Math.ceil((data?.totalRows ?? 0) / ITEMS_PER_PAGE);
+
+  const handlePageChange = (newPage: number) => {
+    if (newPage >= 1 && newPage <= totalPages) {
+      setPage(newPage);
+    }
+  };
+
+  const handleSort = (key: SortKeyType) => {
+    if (key === sortKey) {
+      setSortOrder(sortOrder === SortOrder.ASC ? SortOrder.DESC : SortOrder.ASC);
+    } else {
+      setSortKey(key);
+      setSortOrder(SortOrder.DESC);
+    }
+  };
+
+  const SortableHeader = ({
+    children,
+    sortKey: key,
+  }: {
+    children: React.ReactNode;
+    sortKey: SortKeyType;
+  }) => (
+    <TableHead>
+      <div className="flex justify-end">
+        <Button
+          variant="ghost"
+          onClick={() => handleSort(key)}
+          className="h-8 text-nowrap text-sm font-bold text-text-low-em"
+        >
+          {children}
+          {key === sortKey && (
+            <Icon
+              name={sortOrder === SortOrder.ASC ? 'chevron-up' : 'chevron-down'}
+              className="ml-2 h-4 w-4"
+            />
+          )}
+        </Button>
+      </div>
+    </TableHead>
+  );
```

Comment:
Can we have this component outside the component?

Reply:
Yes and it makes sense! I'll move it to the bottom of the file.